hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/CustomFeesValidator.java

```diff
@@ -228,20 +237,48 @@ private void validateFixedFeeForCreation(
         }
     }
 
+    private void validateFractionalFeeForCreation(TokenType tokenType, CustomFee fee, Set<CustomFee> fees) {
+        validateTrue(isFungibleCommon(tokenType), CUSTOM_FRACTIONAL_FEE_ONLY_ALLOWED_FOR_FUNGIBLE_COMMON);
+
+        final var fractionalFee = fee.fractionalFee();
+        final var numerator = fractionalFee.fractionalAmount().numerator();
+        final var denominator = fractionalFee.fractionalAmount().denominator();
+        final var minimumUnitsToCollect = fractionalFee.minimumAmount();
+        final var nominalMax = fractionalFee.maximumAmount();
+        final var maximumUnitsToCollect = nominalMax == 0 ? Long.MAX_VALUE : nominalMax;
+
+        validateTrue(denominator != 0, FRACTION_DIVIDES_BY_ZERO);
+        validateTrue(numerator > 0 && denominator > 0, CUSTOM_FEE_MUST_BE_POSITIVE);
+        validateTrue(minimumUnitsToCollect >= 0 && maximumUnitsToCollect >= 0, CUSTOM_FEE_MUST_BE_POSITIVE);
+        validateTrue(maximumUnitsToCollect >= minimumUnitsToCollect, FRACTIONAL_FEE_MAX_AMOUNT_LESS_THAN_MIN_AMOUNT);
+
+        fees.add(fee);
+    }
+
     private void validateRoyaltyFeeForCreation(
             @NonNull final TokenType tokenType,
             @NonNull final CustomFee fee,
             @NonNull final ReadableTokenRelationStore tokenRelationStore,
-            @NonNull final WritableTokenStore tokenStore) {
+            @NonNull final WritableTokenStore tokenStore,
+            @NonNull final Set<CustomFee> fees) {
         validateTrue(isNonFungibleUnique(tokenType), CUSTOM_ROYALTY_FEE_ONLY_ALLOWED_FOR_NON_FUNGIBLE_UNIQUE);
         final var royaltyFee = fee.royaltyFeeOrThrow();
+
+        final var numerator = royaltyFee.exchangeValueFraction().numerator();
+        final var denominator = royaltyFee.exchangeValueFraction().denominator();
+        validateTrue(denominator != 0, FRACTION_DIVIDES_BY_ZERO);
+        validateTrue(numerator > 0 && denominator > 0, CUSTOM_FEE_MUST_BE_POSITIVE);
+        validateTrue(numerator <= denominator, ROYALTY_FRACTION_CANNOT_EXCEED_ONE);
+
         if (royaltyFee.hasFallbackFee()) {
+            validateTrue(royaltyFee.fallbackFee().amount() > 0, CUSTOM_FEE_MUST_BE_POSITIVE);
             final var fallbackFee = royaltyFee.fallbackFeeOrThrow();
             if (fallbackFee.hasDenominatingTokenId()) {
                 final var denominatingTokenId = fallbackFee.denominatingTokenIdOrThrow();
                 validateTrue(denominatingTokenId.tokenNum() != 0, CUSTOM_FEE_DENOMINATION_MUST_BE_FUNGIBLE_COMMON);
                 validateExplicitTokenDenomination(
                         fee.feeCollectorAccountId(), denominatingTokenId, tokenRelationStore, tokenStore);
+                fees.add(fee);
```

Comment:
Why are we adding this to fees ? The list of fees is only needed for collector auto association when token mentioned is a sentinel value.

Reply:
Maybe I don't completely understand the business case here. So we need to add to the fees only if the token is a sentinel value(0.0.0) **and** if the custom fee is of type **FIXED_FEE**? Maybe if you can explain a bit or give me resources for that case that would be great 😄 

I will remove the addition of the fees in the two places you mentioned 👍

Reply:
I compare the business logic from `NewRels.listFrom`. In it, we are traversing each custom fee and we check if we should require auto association:
```
for (final var customFee : provisionalToken.getCustomFees()) {
      if (customFee.requiresCollectorAutoAssociation()) {
          final var collector = customFee.getValidatedCollector();
          associateGiven(provisionalToken, collector, tokenStore, associatedSoFar, newRels, dynamicProperties);
      }
  }
```
And `requiresCollectorAutoAssociation` is the following:
```
public boolean requiresCollectorAutoAssociation() {
        switch (feeType) {
            case FRACTIONAL_FEE:
                return true;
            case FIXED_FEE:
                return fixedFeeSpec.usedDenomWildcard();
            case ROYALTY_FEE:
                if (royaltyFeeSpec.hasFallbackFee()) {
                    return royaltyFeeSpec.fallbackFee().usedDenomWildcard();
                }
        }
        return false;
    }
```

* For the **FRACTIONAL_FEE** it's always true - that's why I always add it
* For the **ROYALTY_FEE** when `hasFallbackFee()` is true - that's why I added it inside the `royaltyFee.hasFallbackFee()` (to be more precise maybe I should take it out of the `fallbackFee.hasDenominatingTokenId` if statement)

Does this make sense? Or maybe I am looking at something else 😄

Reply:
I need to look into this comment. But overall makes sense to me. Approving this. I will definitely look into this today.

Reply:
Agree with this change. Thanks for fixing it @vtronkov 🙌